drift/views/v1.py

```diff
@@ -285,6 +287,30 @@ def get_historical_sys_profiles(
                             str(HTTPStatus.FORBIDDEN) + " " + message, request=request
                         )
                         raise HTTPError(HTTPStatus.FORBIDDEN, message=message)
+                    except HTTPError as error:
+                        http_errors.append(error)
+                    except (ItemNotReturned, ServiceError) as error:
+                        other_errors.append(error)
+
+            if http_errors or other_errors:
+                if any(type(other_error) == ServiceError for other_error in other_errors) or any(
```

Comment:
`isinstance()` should be preferred to type checks.

Reply:
Done.